src/platform_impl/macos/mod.rs

```diff
@@ -1 +1,526 @@
+        ns_menu_item
+    }
+
+    pub fn append(&self, item: &dyn crate::MenuEntry) {
+        self.add_menu_item(item, AddOp::Append)
+    }
+
+    pub fn prepend(&self, item: &dyn crate::MenuEntry) {
+        self.add_menu_item(item, AddOp::Insert(0))
+    }
+
+    pub fn insert(&self, item: &dyn crate::MenuEntry, position: usize) {
+        self.add_menu_item(item, AddOp::Insert(position))
+    }
+
+    fn add_menu_item(&self, item: &dyn crate::MenuEntry, op: AddOp) {
+        self.0.borrow_mut().submenu.as_ref().unwrap().add_menu_item(item, op)
+    }
+
+    pub fn remove(&self, item: &dyn crate::MenuEntry) {
+        self.0.borrow_mut().submenu.as_ref().unwrap().remove(item)
+    }
+
+    pub fn items(&self) -> Vec<Box<dyn crate::MenuEntry>> {
+        todo!()
+    }
+
+    pub fn text(&self) -> String {
+        self.0.borrow().text()
+    }
+
+    pub fn set_text(&self, text: &str) {
+        self.0.borrow_mut().set_text(text)
+    }
+
+    pub fn is_enabled(&self) -> bool {
+        self.0.borrow().is_enabled()
+    }
+
+    pub fn set_enabled(&self, enabled: bool) {
+        self.0.borrow_mut().set_enabled(enabled)
+    }
+}
+
+#[derive(Clone, Debug)]
+pub(crate) struct MenuItem(Rc<RefCell<MenuChild>>);
+
+impl MenuItem {
+    pub fn new(text: &str, enabled: bool, accelerator: Option<Accelerator>) -> Self {
+        Self(Rc::new(RefCell::new(MenuChild {
+            type_: MenuEntryType::Normal,
+            text: text.to_string().replace("&", ""),
+            enabled,
+            id: COUNTER.next(),
+            accelerator,
+            ..Default::default()
+        })))
+    }
+
+    pub fn ns_item_for_menu(&self, menu: &Menu) -> id {
+        let mut child = self.0.as_ref().borrow_mut();
+
+        if let Some(&ns_menu_item) = child.ns_menuitems.get(menu) {
+            return ns_menu_item;
+        }
+
+        let ns_menu_item = create_ns_menu_item(&child.text, sel!(fireMenubarAction:), &child.accelerator);
+
+        if !child.enabled {
+            unsafe {
+                let () = msg_send![ns_menu_item, setEnabled: NO];
+            }
+        }
+
+        child.ns_menuitems.insert(menu.clone(), ns_menu_item);
+        ns_menu_item
+    }
+
+    pub fn id(&self) -> u32 {
+        self.0.borrow().id()
+    }
+
+    pub fn text(&self) -> String {
+        self.0.borrow().text()
+    }
+
+    pub fn set_text(&self, text: &str) {
+        self.0.borrow_mut().set_text(text)
+    }
+
+    pub fn is_enabled(&self) -> bool {
+        self.0.borrow().is_enabled()
+    }
+
+    pub fn set_enabled(&self, enabled: bool) {
+        self.0.borrow_mut().set_enabled(enabled)
+    }
+}
+
+#[derive(Clone, Debug)]
+pub(crate) struct PredefinedMenuItem(Rc<RefCell<MenuChild>>);
+
+impl PredefinedMenuItem {
+    pub fn new(item_type: PredfinedMenuItemType, text: Option<String>) -> Self {
+        let text = text
+            .unwrap_or_else(|| item_type.text().to_string())
+            .replace("&", "");
```

Comment:
this will remove all `&`, we should only remove single `&` and keep `&&` and convert it to a single `&`

The rule is, `&` before a char defines an auto mnemonic for this menu item, if you want to display `&` as is, you should escape it and use `&&`

Reply:
Ah yeah I did wonder about that. I guess some regex will be needed 😑

Reply:
Do the 'mnemonics' work on Windows btw? I couldn't find any evidence of them when I ran the example on Win 11.

Reply:
Yeah they do, press Alt and you should see a character getting underline

Reply:
> Ah yeah I did wonder about that. I guess some regex will be needed 😑

No need for regex, take a look at a dirty way to doing it in gtk/accelerator.ra

Reply:
> Yeah they do, press Alt and you should see a character getting underline

Just tried again and I'm definitely not seeing that in my Windows 11 VM, all that happens is when I release Alt the menu closes.

Reply:
I will double check that but mnemonics are not supported on macOS iirc, so you only need to remove single `&` and convert `&&` to `&`

Reply:
There is definitely something weird with the menubar mnemonics but you can press `Alt+F` to open the file submenu and you should see the menu items have the mnemonics displayed correctly. I will see if I can fix the menubar.

Reply:
The menubar issue is something that needs to be fixed in tao and winit eventloop. I will open a PR to fix it. Thanks for catching this.

Reply:
Ah, that's great that you've tracked it down!

> mnemonics are not supported on macOS iirc, so you only need to remove single `&` and convert `&&` to `&`

Yup that's right, macOS doesn't support them anymore.